python/tests/test_umap.py

```diff
@@ -493,3 +549,50 @@ def assert_umap_model(model: UMAPModel) -> None:
         trust_diff = loc_umap - dist_umap
 
         assert trust_diff <= 0.15
+
+
+@pytest.mark.parametrize("n_rows", [3000])
+@pytest.mark.parametrize("n_cols", [64, 128])
+@pytest.mark.parametrize("nnz", [7, 12])
+def test_umap_sparse_vector(
+    n_rows: int, n_cols: int, nnz: int, gpu_number: int, tmp_path: str
+) -> None:
+    import pyspark
```

Comment:
Better to add a new test case where a csr_matrix gets split in estimator class and reconstructed in model class.

Reply:
This was added to the model persistence test.